src/Symfony/Component/Serializer/Mapping/Loader/AnnotationLoader.php

```diff
@@ -157,7 +157,18 @@ public function loadAnnotations(object $reflector): iterable
     {
         foreach ($reflector->getAttributes() as $attribute) {
             if ($this->isKnownAttribute($attribute->getName())) {
-                yield $attribute->newInstance();
+                try {
+                    yield $attribute->newInstance();
+                } catch (\Error $e) {
+                    $on = match (true) {
+                        $reflector instanceof \ReflectionClass => ' on class '.$reflector->name,
+                        $reflector instanceof \ReflectionMethod => sprintf(' on "%s::%s()"', $reflector->getDeclaringClass()->name, $reflector->name),
+                        $reflector instanceof \ReflectionProperty => sprintf(' on "%s::$%s"', $reflector->getDeclaringClass()->name, $reflector->name),
+                        default => '',
+                    };
+
+                    throw new MappingException(sprintf('Could not instantiate attribute "%s"%s.', $attribute->getName(), $on), 0, $e);
```

Comment:
shouldn't we add $e->getMessage() at the end of the message?

Reply:
The new exception already wrap the previous one. You already made me add the attribute (which is redundant too). you can see the initial screenshot in the PR, all information are here to debug the situation I think

Reply:
OK, works for me.